src/REP15Utils.sol

```diff
@@ -0,0 +1,38 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.26;
+
+library REP15Utils {
+  struct Delegation {
+    address delegatee;
+    uint64 until;
+    bool delegated;
+  }
+
+  struct Context {
+    address controller;
+    uint64 detachingDuration;
+  }
+
+  struct TokenContext {
+    bool attached;
+    bool locked;
+    address user;
+    uint64 readyForDetachmentAt;
+  }
+
+  function isActive(Delegation storage self) internal view returns (bool) {
+    return self.delegated && self.until > block.timestamp;
+  }
+
+  function isPending(Delegation storage self) internal view returns (bool) {
+    return !self.delegated && self.until > block.timestamp;
+  }
+
+  function isExistent(Context storage self) internal view returns (bool) {
```

Comment:
Should not passing storage pointer into view function
```suggestion
  function isExistent(Context memory self) internal view returns (bool) {
```

Reply:
Your code costs more gas.

Reply:
Store 1 slot into memory cost only 3 gas. Bring storage pointer to everywhere just for saving little gas makes no sense to me

Reply:
I see no problem using a storage pointer in a view function. Many libraries use storage pointers even in view functions (e.g., [Checkpoints](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/05f218fb6617932e56bf5388c3b389c3028a7b73/contracts/utils/structs/Checkpoints.sol#L115), [EnumerableSet](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/05f218fb6617932e56bf5388c3b389c3028a7b73/contracts/utils/structs/EnumerableSet.sol#L141)). Additionally, this saves much gas, not only 3! Let's benchmark.